fireflower/targets.py

```diff
@@ -195,3 +205,71 @@ def read_csv(self, **kwargs):
                     return pd.read_csv(g, **kwargs)
             else:
                 return pd.read_csv(f, **kwargs)
+
+
+class S3TypedCSVTarget(S3CSVTarget):
+    def __init__(self, path, types, compressed=True,
+                 kwargs_in=None, kwargs_out=None, format=None):
+        self.types = types
+        if compressed:
+            format = luigi.format.Nop
+
+        super(S3TypedCSVTarget, self).__init__(path, compressed,
+                                               kwargs_in, kwargs_out, format)
+
+    def write_typed_csv(self, df, **kwargs):
+        if self.kwargs_out:
+            kwargs = toolz.merge(self.kwargs_out, kwargs)
+        with self._open_writer() as f:
+            if self.compressed:
+                with TextIOWrapper(GzipFile(fileobj=f, mode='wb')) as g:
+                    transformed = pd.DataFrame.from_items(
+                            (colname,
+                                self.types[colname].output(col)
+                                if colname in self.types else col)
+                            for colname, col in df.items())
+
+                    transformed.to_csv(g, compression='gzip', **kwargs)
+            else:
+                df.to_csv(f, **kwargs)
+
+    def read_typed_csv(self, **kwargs):
+        if self.kwargs_in:
+            kwargs = toolz.merge(self.kwargs_in, kwargs)
+        with self._open_reader() as f:
+            dtype = {colname: coltype.serialization_dtype
+                     for colname, coltype in self.types.items()}
+            if self.compressed:
+                df = pd.read_csv(filepath_or_buffer=f,
```

Comment:
can you read this the same way we read in S3CSVTarget?

 with TextIOWrapper(GzipFile(fileobj=f, mode='rb')) as g: